packages/frontend-plugin-api/src/wiring/createPlugin.test.ts

```diff
@@ -250,4 +271,68 @@ describe('createPlugin', () => {
       }),
     ).toThrow("Plugin 'test' provided duplicate extensions: test/2, test/3");
   });
+
+  describe('overrides', () => {
+    it('should return a plugin instance with the correct namespace', () => {
+      const plugin = createPlugin({
+        id: 'test',
+        extensions: [Extension1, Extension2],
+      });
+
+      expect(plugin.getExtension('test/1')).toMatchInlineSnapshot(`
+        {
+          "$$type": "@backstage/ExtensionDefinition",
```

Comment:
Hmm I wonder if we can do this without having to have an override property. 

If theres two ID's and one is provided by discovery and one is installed in the app using DI then we can mark the one installed by DI as overrides already in the `app-viz` and making it clear that this is one is overriding the discovered one, or something like that at least?

> when I install two packages that override two different extensions of the same plugin

I'm not sure we would support this, right @Rugvip? You'd only do the override once to add replace an extension in the `catalog` plugin for instance I think, I'm not sure we'd have two different plugins with overriding the same one?

Reply:
I want to add a `description` field or similar to the internal type that describes where the particular extension is defined, including the override. Similar to route refs: https://github.com/backstage/backstage/blob/5b665a3810a45987bc9b44e74b0fb14de4d66fe8/packages/frontend-plugin-api/src/routing/RouteRef.ts#L134

I'm thinking that would cover all use-cases when it comes to identifying exactly which extension is installed and active in the app. That said I think it would be pretty sweet to just have an "overridden" marker in the UI too, but not a requirement for now imo.

Reply:
> I'm not sure we would support this, right @Rugvip? You'd only do the override once to add replace an extension in the catalog plugin for instance I think, I'm not sure we'd have two different plugins with overriding the same one?

I was thinking on this example: install a community package that overrides the app theme extensions, and another community package that overrides the app sidebar. So there will be two different packages overriding the same plugin. Would that be a valid case?
Another example could be a package overriding the Search plugin to create a custom search result blueprint and another package overriding the Search plugin just for providing a custom advanced search page.